mlir/include/mlir/IR/OperationSupport.h

```diff
@@ -1131,6 +1131,10 @@ class OpPrintingFlags {
   /// elements.
   OpPrintingFlags &elideLargeElementsAttrs(int64_t largeElementLimit = 16);
 
+  /// Enable or disable printing of large element attributes as a hex string. By
+  /// default, printing as hex is allowed (enabled).
+  OpPrintingFlags &allowPrintingElementsAttrWithHex(bool allowHex = true);
```

Comment:
this patch is similar to the original we have in the downstream. however, looking at this now, I wonder if it makes more sense to expose `clOptions->printElementsAttrWithHexIfLarger` threshold here (so having an int64_t value instead of a bool). Then `-1` would act as "disable"

Reply:
Considering we already have *printElementsAttrWithHexIfLarger* as part of the cl::opt, I would assume all you want is make this available as a C++ API indeed instead of adding another flag here.

Reply:
done.